wordpress-mobile/WordPress-iOS
# Post Formats: Add refresh control and sync for post formats.

Fixes #2573 (bonus addition)

Just as in https://github.com/wordpress-mobile/WordPress-iOS/pull/6451 we're adding a `UIRefreshControl` for refreshing the post formats. While these don't change very often, the same problem is present. If a user _were_ changing post formats (themes) while trying to use the editor, they wouldn't be able to pull in the current post formats without a hard sync.

So now we allow a refresh, via an addition to `SettingsSelectionViewController` that allows for passing in an `onRefresh` block that then sets up a `UIRefreshControl`.

To test:
1. Log in to a .COM account.
2. Open the Editor > Options > Post Format.
3. Notice the current list of post formats.
4. On the web, activate a theme with limited post formats, such as: https://theme.wordpress.com/themes/libre-2/
5. Back in the app, pull down to refresh the post formats.
6. Check that the list of post formats matches the web. (only Standard in the case of Libre 2)
7. Switch back to another theme with more post formats: https://en.support.wordpress.com/post-formats/
8. Refresh the Post Format table in-app with the new theme, the post formats should match the available formats from the theme's list.

To test:
1. Log out.
2. Add a self-hosted site.
3. Open the Editor > Options > Post Format.
4. Notice the current list of post formats.
5. On the web, activate a theme with limited post formats, such as: https://wordpress.org/themes/level/
6. Check that the list of post formats matches the web.
7. Switch back to another theme with more post formats, such as the default theme.
8. Refresh the Post Format table in-app with the new theme, the post formats should match the available formats for the theme.

Needs review: @nheagy can you take a shot at this one?
